modules/options/src/main/scala/scala/build/interactive/Interactive.scala

```diff
@@ -0,0 +1,77 @@
+package scala.build.interactive
+
+import scala.io.StdIn.readLine
+
+sealed abstract class Interactive extends Product with Serializable {
+  def confirmOperation(msg: String): Option[Boolean]                = None
+  def chooseOne(msg: String, options: List[String]): Option[String] = None
+}
+
+object Interactive {
+
+  case object InteractiveNop extends Interactive
+
+  case object InteractiveAsk extends Interactive {
+
+    sealed abstract class Action[V] extends Product with Serializable {
+      def msg: String
+      def action: Option[V]
+      final def run: Option[V] =
+        if (coursier.paths.Util.useAnsiOutput())
+          action
+        else None
+    }
+
+    case class ConfirmOperation(msg: String) extends Action[Boolean] {
+      override def action: Option[Boolean] = {
+        System.err.println(s"$msg [Y/n]")
```

Comment:
Should we favor using a logger here, rather than just stderr?

Reply:
nvm, I see [why not](https://github.com/VirtusLab/scala-cli/pull/1016#discussion_r875133009)